framework/utils/CDateTimeParser.php

```diff
@@ -83,6 +83,9 @@ public static function parse($value,$pattern='MM/dd/yyyy',$defaults=array())
 		if(self::$_mbstringAvailable===null)
 			self::$_mbstringAvailable=extension_loaded('mbstring');
 
+		if($value===null)
```

Comment:
`$value` needs to be string according to phpdoc. I think it makes more sense to handle null/empty string directly in validators instead of here.

Reply:
This is the same issue as https://github.com/yiisoft/yii/pull/4503#discussion_r1054552182

This is also a case where the tests are passing `null` and `false` into a method that only accepts `$value` as a string. https://github.com/yiisoft/yii/blob/1c6f6fe221404de9afeb32059e6671676237482c/tests/framework/utils/CDateTimeParserTest.php#L196-L197

I looked for all the usages of `CDateTimeParser::parse` and it appers that `$value` is always checked for `null` before being passed to `CDateTimeParser::parse`: https://github.com/yiisoft/yii/blob/1c6f6fe221404de9afeb32059e6671676237482c/framework/validators/CTypeValidator.php#L123-L128
and https://github.com/yiisoft/yii/blob/1c6f6fe221404de9afeb32059e6671676237482c/framework/validators/CDateValidator.php#L52

These invalid assertions should be safe to remove but I'll wait for https://github.com/yiisoft/yii/pull/4503#discussion_r1054571980 in the event that I am missing something.